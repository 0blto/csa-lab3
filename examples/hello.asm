section data
    word: 'Hello world!'

section text
start:
    mov @r2 0
    printer:
        ld @r1 (word|r2)
        cmp @r1 0
        jz :fin
        st !stdout @r1
        inc @r2
        jmp :printer
    fin:
        hlt