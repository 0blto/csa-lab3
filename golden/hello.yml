in_source: |-
  section data
      word: 'Hello world!'
  
  section text
  start:
      mov @r2 0
      printer:
          ld @r1 (word|r2)
          cmp @r1 0
          jz :fin
          st !stdout @r1
          inc @r2
          jmp :printer
      fin:
          hlt
in_stdin: |
  xxxnsane!
out_code: |
  [{"index": 2, "instruction": "data", "arguments": [{"address_mode": "data", "data": 72, "offset": 0}, {"address_mode": "data", "data": 101, "offset": 0}, {"address_mode": "data", "data": 108, "offset": 0}, {"address_mode": "data", "data": 108, "offset": 0}, {"address_mode": "data", "data": 111, "offset": 0}, {"address_mode": "data", "data": 32, "offset": 0}, {"address_mode": "data", "data": 119, "offset": 0}, {"address_mode": "data", "data": 111, "offset": 0}, {"address_mode": "data", "data": 114, "offset": 0}, {"address_mode": "data", "data": 108, "offset": 0}, {"address_mode": "data", "data": 100, "offset": 0}, {"address_mode": "data", "data": 33, "offset": 0}, {"address_mode": "data", "data": 0, "offset": 0}]},
   {"index": 0, "instruction": "mov", "arguments": [{"address_mode": "register", "data": "r2", "offset": 0}, {"address_mode": "data", "data": 0, "offset": 0}]},
   {"index": 1, "instruction": "ld", "arguments": [{"address_mode": "register", "data": "r1", "offset": 0}, {"address_mode": "indirect", "data": 2, "offset": "r2"}]},
   {"index": 2, "instruction": "cmp", "arguments": [{"address_mode": "register", "data": "r1", "offset": 0}, {"address_mode": "data", "data": 0, "offset": 0}]},
   {"index": 3, "instruction": "jz", "arguments": [{"address_mode": "absolute", "data": 7, "offset": 0}]},
   {"index": 4, "instruction": "st", "arguments": [{"address_mode": "absolute", "data": 1, "offset": 0}, {"address_mode": "register", "data": "r1", "offset": 0}]},
   {"index": 5, "instruction": "inc", "arguments": [{"address_mode": "register", "data": "r2", "offset": 0}]},
   {"index": 6, "instruction": "jmp", "arguments": [{"address_mode": "absolute", "data": 1, "offset": 0}]},
   {"index": 7, "instruction": "hlt", "arguments": []}]

out_stdout: |
  source LoC: 11 code instr: 8
  Instructions number: 76
  72[H] 101[e] 108[l] 108[l] 111[o] 32[ ] 119[w] 111[o] 114[r] 108[l] 100[d] 33[!]

out_log: |
  INFO    PC: 0, ADDR_BUS: 0, R1: 0, R2: 0, R3: 0, R4: 0, R5: 0, R6: 0, D_ALU_BUD: 0, A_ALU_BUD: 0, MEM_BUS: 0, N: False, Z: False 
  INFO    PC: 1, ADDR_BUS: 0, R1: 0, R2: 0, R3: 0, R4: 0, R5: 0, R6: 0, D_ALU_BUD: 0, A_ALU_BUD: 0, MEM_BUS: 0, N: False, Z: False INDEX: 0, OPCODE: MOV, ARGS: [(register, r2, 0), (data, 0, 0)]
  INFO    PC: 2, ADDR_BUS: 2, R1: 72, R2: 0, R3: 0, R4: 0, R5: 0, R6: 0, D_ALU_BUD: 0, A_ALU_BUD: 2, MEM_BUS: 72, N: False, Z: False INDEX: 1, OPCODE: LD, ARGS: [(register, r1, 0), (indirect, 2, r2)]
  INFO    PC: 3, ADDR_BUS: 2, R1: 72, R2: 0, R3: 0, R4: 0, R5: 0, R6: 0, D_ALU_BUD: 72, A_ALU_BUD: 2, MEM_BUS: 72, N: False, Z: False INDEX: 2, OPCODE: CMP, ARGS: [(register, r1, 0), (data, 0, 0)]
  INFO    PC: 4, ADDR_BUS: 2, R1: 72, R2: 0, R3: 0, R4: 0, R5: 0, R6: 0, D_ALU_BUD: 72, A_ALU_BUD: 2, MEM_BUS: 72, N: False, Z: False INDEX: 3, OPCODE: JZ, ARGS: [(absolute, 7, 0)]
  INFO    PC: 5, ADDR_BUS: 1, R1: 72, R2: 0, R3: 0, R4: 0, R5: 0, R6: 0, D_ALU_BUD: 72, A_ALU_BUD: 2, MEM_BUS: 72, N: False, Z: False INDEX: 4, OPCODE: ST, ARGS: [(absolute, 1, 0), (register, r1, 0)]
  INFO    PC: 6, ADDR_BUS: 1, R1: 72, R2: 1, R3: 0, R4: 0, R5: 0, R6: 0, D_ALU_BUD: 1, A_ALU_BUD: 2, MEM_BUS: 72, N: False, Z: False INDEX: 5, OPCODE: INC, ARGS: [(register, r2, 0)]
  INFO    PC: 1, ADDR_BUS: 1, R1: 72, R2: 1, R3: 0, R4: 0, R5: 0, R6: 0, D_ALU_BUD: 1, A_ALU_BUD: 2, MEM_BUS: 72, N: False, Z: False INDEX: 6, OPCODE: JMP, ARGS: [(absolute, 1, 0)]
  INFO    PC: 2, ADDR_BUS: 3, R1: 101, R2: 1, R3: 0, R4: 0, R5: 0, R6: 0, D_ALU_BUD: 1, A_ALU_BUD: 3, MEM_BUS: 101, N: False, Z: False INDEX: 1, OPCODE: LD, ARGS: [(register, r1, 0), (indirect, 2, r2)]
  INFO    PC: 3, ADDR_BUS: 3, R1: 101, R2: 1, R3: 0, R4: 0, R5: 0, R6: 0, D_ALU_BUD: 101, A_ALU_BUD: 3, MEM_BUS: 101, N: False, Z: False INDEX: 2, OPCODE: CMP, ARGS: [(register, r1, 0), (data, 0, 0)]
  INFO    PC: 4, ADDR_BUS: 3, R1: 101, R2: 1, R3: 0, R4: 0, R5: 0, R6: 0, D_ALU_BUD: 101, A_ALU_BUD: 3, MEM_BUS: 101, N: False, Z: False INDEX: 3, OPCODE: JZ, ARGS: [(absolute, 7, 0)]
  INFO    PC: 5, ADDR_BUS: 1, R1: 101, R2: 1, R3: 0, R4: 0, R5: 0, R6: 0, D_ALU_BUD: 101, A_ALU_BUD: 3, MEM_BUS: 101, N: False, Z: False INDEX: 4, OPCODE: ST, ARGS: [(absolute, 1, 0), (register, r1, 0)]
  INFO    PC: 6, ADDR_BUS: 1, R1: 101, R2: 2, R3: 0, R4: 0, R5: 0, R6: 0, D_ALU_BUD: 2, A_ALU_BUD: 3, MEM_BUS: 101, N: False, Z: False INDEX: 5, OPCODE: INC, ARGS: [(register, r2, 0)]
  INFO    PC: 1, ADDR_BUS: 1, R1: 101, R2: 2, R3: 0, R4: 0, R5: 0, R6: 0, D_ALU_BUD: 2, A_ALU_BUD: 3, MEM_BUS: 101, N: False, Z: False INDEX: 6, OPCODE: JMP, ARGS: [(absolute, 1, 0)]
  INFO    PC: 2, ADDR_BUS: 4, R1: 108, R2: 2, R3: 0, R4: 0, R5: 0, R6: 0, D_ALU_BUD: 2, A_ALU_BUD: 4, MEM_BUS: 108, N: False, Z: False INDEX: 1, OPCODE: LD, ARGS: [(register, r1, 0), (indirect, 2, r2)]
  INFO    PC: 3, ADDR_BUS: 4, R1: 108, R2: 2, R3: 0, R4: 0, R5: 0, R6: 0, D_ALU_BUD: 108, A_ALU_BUD: 4, MEM_BUS: 108, N: False, Z: False INDEX: 2, OPCODE: CMP, ARGS: [(register, r1, 0), (data, 0, 0)]
  INFO    PC: 4, ADDR_BUS: 4, R1: 108, R2: 2, R3: 0, R4: 0, R5: 0, R6: 0, D_ALU_BUD: 108, A_ALU_BUD: 4, MEM_BUS: 108, N: False, Z: False INDEX: 3, OPCODE: JZ, ARGS: [(absolute, 7, 0)]
  INFO    PC: 5, ADDR_BUS: 1, R1: 108, R2: 2, R3: 0, R4: 0, R5: 0, R6: 0, D_ALU_BUD: 108, A_ALU_BUD: 4, MEM_BUS: 108, N: False, Z: False INDEX: 4, OPCODE: ST, ARGS: [(absolute, 1, 0), (register, r1, 0)]
  INFO    PC: 6, ADDR_BUS: 1, R1: 108, R2: 3, R3: 0, R4: 0, R5: 0, R6: 0, D_ALU_BUD: 3, A_ALU_BUD: 4, MEM_BUS: 108, N: False, Z: False INDEX: 5, OPCODE: INC, ARGS: [(register, r2, 0)]
  INFO    PC: 1, ADDR_BUS: 1, R1: 108, R2: 3, R3: 0, R4: 0, R5: 0, R6: 0, D_ALU_BUD: 3, A_ALU_BUD: 4, MEM_BUS: 108, N: False, Z: False INDEX: 6, OPCODE: JMP, ARGS: [(absolute, 1, 0)]
  INFO    PC: 2, ADDR_BUS: 5, R1: 108, R2: 3, R3: 0, R4: 0, R5: 0, R6: 0, D_ALU_BUD: 3, A_ALU_BUD: 5, MEM_BUS: 108, N: False, Z: False INDEX: 1, OPCODE: LD, ARGS: [(register, r1, 0), (indirect, 2, r2)]
  INFO    PC: 3, ADDR_BUS: 5, R1: 108, R2: 3, R3: 0, R4: 0, R5: 0, R6: 0, D_ALU_BUD: 108, A_ALU_BUD: 5, MEM_BUS: 108, N: False, Z: False INDEX: 2, OPCODE: CMP, ARGS: [(register, r1, 0), (data, 0, 0)]
  INFO    PC: 4, ADDR_BUS: 5, R1: 108, R2: 3, R3: 0, R4: 0, R5: 0, R6: 0, D_ALU_BUD: 108, A_ALU_BUD: 5, MEM_BUS: 108, N: False, Z: False INDEX: 3, OPCODE: JZ, ARGS: [(absolute, 7, 0)]
  INFO    PC: 5, ADDR_BUS: 1, R1: 108, R2: 3, R3: 0, R4: 0, R5: 0, R6: 0, D_ALU_BUD: 108, A_ALU_BUD: 5, MEM_BUS: 108, N: False, Z: False INDEX: 4, OPCODE: ST, ARGS: [(absolute, 1, 0), (register, r1, 0)]
  INFO    PC: 6, ADDR_BUS: 1, R1: 108, R2: 4, R3: 0, R4: 0, R5: 0, R6: 0, D_ALU_BUD: 4, A_ALU_BUD: 5, MEM_BUS: 108, N: False, Z: False INDEX: 5, OPCODE: INC, ARGS: [(register, r2, 0)]
  INFO    PC: 1, ADDR_BUS: 1, R1: 108, R2: 4, R3: 0, R4: 0, R5: 0, R6: 0, D_ALU_BUD: 4, A_ALU_BUD: 5, MEM_BUS: 108, N: False, Z: False INDEX: 6, OPCODE: JMP, ARGS: [(absolute, 1, 0)]
  INFO    PC: 2, ADDR_BUS: 6, R1: 111, R2: 4, R3: 0, R4: 0, R5: 0, R6: 0, D_ALU_BUD: 4, A_ALU_BUD: 6, MEM_BUS: 111, N: False, Z: False INDEX: 1, OPCODE: LD, ARGS: [(register, r1, 0), (indirect, 2, r2)]
  INFO    PC: 3, ADDR_BUS: 6, R1: 111, R2: 4, R3: 0, R4: 0, R5: 0, R6: 0, D_ALU_BUD: 111, A_ALU_BUD: 6, MEM_BUS: 111, N: False, Z: False INDEX: 2, OPCODE: CMP, ARGS: [(register, r1, 0), (data, 0, 0)]
  INFO    PC: 4, ADDR_BUS: 6, R1: 111, R2: 4, R3: 0, R4: 0, R5: 0, R6: 0, D_ALU_BUD: 111, A_ALU_BUD: 6, MEM_BUS: 111, N: False, Z: False INDEX: 3, OPCODE: JZ, ARGS: [(absolute, 7, 0)]
  INFO    PC: 5, ADDR_BUS: 1, R1: 111, R2: 4, R3: 0, R4: 0, R5: 0, R6: 0, D_ALU_BUD: 111, A_ALU_BUD: 6, MEM_BUS: 111, N: False, Z: False INDEX: 4, OPCODE: ST, ARGS: [(absolute, 1, 0), (register, r1, 0)]
  INFO    PC: 6, ADDR_BUS: 1, R1: 111, R2: 5, R3: 0, R4: 0, R5: 0, R6: 0, D_ALU_BUD: 5, A_ALU_BUD: 6, MEM_BUS: 111, N: False, Z: False INDEX: 5, OPCODE: INC, ARGS: [(register, r2, 0)]
  INFO    PC: 1, ADDR_BUS: 1, R1: 111, R2: 5, R3: 0, R4: 0, R5: 0, R6: 0, D_ALU_BUD: 5, A_ALU_BUD: 6, MEM_BUS: 111, N: False, Z: False INDEX: 6, OPCODE: JMP, ARGS: [(absolute, 1, 0)]
  INFO    PC: 2, ADDR_BUS: 7, R1: 32, R2: 5, R3: 0, R4: 0, R5: 0, R6: 0, D_ALU_BUD: 5, A_ALU_BUD: 7, MEM_BUS: 32, N: False, Z: False INDEX: 1, OPCODE: LD, ARGS: [(register, r1, 0), (indirect, 2, r2)]
  INFO    PC: 3, ADDR_BUS: 7, R1: 32, R2: 5, R3: 0, R4: 0, R5: 0, R6: 0, D_ALU_BUD: 32, A_ALU_BUD: 7, MEM_BUS: 32, N: False, Z: False INDEX: 2, OPCODE: CMP, ARGS: [(register, r1, 0), (data, 0, 0)]
  INFO    PC: 4, ADDR_BUS: 7, R1: 32, R2: 5, R3: 0, R4: 0, R5: 0, R6: 0, D_ALU_BUD: 32, A_ALU_BUD: 7, MEM_BUS: 32, N: False, Z: False INDEX: 3, OPCODE: JZ, ARGS: [(absolute, 7, 0)]
  INFO    PC: 5, ADDR_BUS: 1, R1: 32, R2: 5, R3: 0, R4: 0, R5: 0, R6: 0, D_ALU_BUD: 32, A_ALU_BUD: 7, MEM_BUS: 32, N: False, Z: False INDEX: 4, OPCODE: ST, ARGS: [(absolute, 1, 0), (register, r1, 0)]
  INFO    PC: 6, ADDR_BUS: 1, R1: 32, R2: 6, R3: 0, R4: 0, R5: 0, R6: 0, D_ALU_BUD: 6, A_ALU_BUD: 7, MEM_BUS: 32, N: False, Z: False INDEX: 5, OPCODE: INC, ARGS: [(register, r2, 0)]
  INFO    PC: 1, ADDR_BUS: 1, R1: 32, R2: 6, R3: 0, R4: 0, R5: 0, R6: 0, D_ALU_BUD: 6, A_ALU_BUD: 7, MEM_BUS: 32, N: False, Z: False INDEX: 6, OPCODE: JMP, ARGS: [(absolute, 1, 0)]
  INFO    PC: 2, ADDR_BUS: 8, R1: 119, R2: 6, R3: 0, R4: 0, R5: 0, R6: 0, D_ALU_BUD: 6, A_ALU_BUD: 8, MEM_BUS: 119, N: False, Z: False INDEX: 1, OPCODE: LD, ARGS: [(register, r1, 0), (indirect, 2, r2)]
  INFO    PC: 3, ADDR_BUS: 8, R1: 119, R2: 6, R3: 0, R4: 0, R5: 0, R6: 0, D_ALU_BUD: 119, A_ALU_BUD: 8, MEM_BUS: 119, N: False, Z: False INDEX: 2, OPCODE: CMP, ARGS: [(register, r1, 0), (data, 0, 0)]
  INFO    PC: 4, ADDR_BUS: 8, R1: 119, R2: 6, R3: 0, R4: 0, R5: 0, R6: 0, D_ALU_BUD: 119, A_ALU_BUD: 8, MEM_BUS: 119, N: False, Z: False INDEX: 3, OPCODE: JZ, ARGS: [(absolute, 7, 0)]
  INFO    PC: 5, ADDR_BUS: 1, R1: 119, R2: 6, R3: 0, R4: 0, R5: 0, R6: 0, D_ALU_BUD: 119, A_ALU_BUD: 8, MEM_BUS: 119, N: False, Z: False INDEX: 4, OPCODE: ST, ARGS: [(absolute, 1, 0), (register, r1, 0)]
  INFO    PC: 6, ADDR_BUS: 1, R1: 119, R2: 7, R3: 0, R4: 0, R5: 0, R6: 0, D_ALU_BUD: 7, A_ALU_BUD: 8, MEM_BUS: 119, N: False, Z: False INDEX: 5, OPCODE: INC, ARGS: [(register, r2, 0)]
  INFO    PC: 1, ADDR_BUS: 1, R1: 119, R2: 7, R3: 0, R4: 0, R5: 0, R6: 0, D_ALU_BUD: 7, A_ALU_BUD: 8, MEM_BUS: 119, N: False, Z: False INDEX: 6, OPCODE: JMP, ARGS: [(absolute, 1, 0)]
  INFO    PC: 2, ADDR_BUS: 9, R1: 111, R2: 7, R3: 0, R4: 0, R5: 0, R6: 0, D_ALU_BUD: 7, A_ALU_BUD: 9, MEM_BUS: 111, N: False, Z: False INDEX: 1, OPCODE: LD, ARGS: [(register, r1, 0), (indirect, 2, r2)]
  INFO    PC: 3, ADDR_BUS: 9, R1: 111, R2: 7, R3: 0, R4: 0, R5: 0, R6: 0, D_ALU_BUD: 111, A_ALU_BUD: 9, MEM_BUS: 111, N: False, Z: False INDEX: 2, OPCODE: CMP, ARGS: [(register, r1, 0), (data, 0, 0)]
  INFO    PC: 4, ADDR_BUS: 9, R1: 111, R2: 7, R3: 0, R4: 0, R5: 0, R6: 0, D_ALU_BUD: 111, A_ALU_BUD: 9, MEM_BUS: 111, N: False, Z: False INDEX: 3, OPCODE: JZ, ARGS: [(absolute, 7, 0)]
  INFO    PC: 5, ADDR_BUS: 1, R1: 111, R2: 7, R3: 0, R4: 0, R5: 0, R6: 0, D_ALU_BUD: 111, A_ALU_BUD: 9, MEM_BUS: 111, N: False, Z: False INDEX: 4, OPCODE: ST, ARGS: [(absolute, 1, 0), (register, r1, 0)]
  INFO    PC: 6, ADDR_BUS: 1, R1: 111, R2: 8, R3: 0, R4: 0, R5: 0, R6: 0, D_ALU_BUD: 8, A_ALU_BUD: 9, MEM_BUS: 111, N: False, Z: False INDEX: 5, OPCODE: INC, ARGS: [(register, r2, 0)]
  INFO    PC: 1, ADDR_BUS: 1, R1: 111, R2: 8, R3: 0, R4: 0, R5: 0, R6: 0, D_ALU_BUD: 8, A_ALU_BUD: 9, MEM_BUS: 111, N: False, Z: False INDEX: 6, OPCODE: JMP, ARGS: [(absolute, 1, 0)]
  INFO    PC: 2, ADDR_BUS: 10, R1: 114, R2: 8, R3: 0, R4: 0, R5: 0, R6: 0, D_ALU_BUD: 8, A_ALU_BUD: 10, MEM_BUS: 114, N: False, Z: False INDEX: 1, OPCODE: LD, ARGS: [(register, r1, 0), (indirect, 2, r2)]
  INFO    PC: 3, ADDR_BUS: 10, R1: 114, R2: 8, R3: 0, R4: 0, R5: 0, R6: 0, D_ALU_BUD: 114, A_ALU_BUD: 10, MEM_BUS: 114, N: False, Z: False INDEX: 2, OPCODE: CMP, ARGS: [(register, r1, 0), (data, 0, 0)]
  INFO    PC: 4, ADDR_BUS: 10, R1: 114, R2: 8, R3: 0, R4: 0, R5: 0, R6: 0, D_ALU_BUD: 114, A_ALU_BUD: 10, MEM_BUS: 114, N: False, Z: False INDEX: 3, OPCODE: JZ, ARGS: [(absolute, 7, 0)]
  INFO    PC: 5, ADDR_BUS: 1, R1: 114, R2: 8, R3: 0, R4: 0, R5: 0, R6: 0, D_ALU_BUD: 114, A_ALU_BUD: 10, MEM_BUS: 114, N: False, Z: False INDEX: 4, OPCODE: ST, ARGS: [(absolute, 1, 0), (register, r1, 0)]
  INFO    PC: 6, ADDR_BUS: 1, R1: 114, R2: 9, R3: 0, R4: 0, R5: 0, R6: 0, D_ALU_BUD: 9, A_ALU_BUD: 10, MEM_BUS: 114, N: False, Z: False INDEX: 5, OPCODE: INC, ARGS: [(register, r2, 0)]
  INFO    PC: 1, ADDR_BUS: 1, R1: 114, R2: 9, R3: 0, R4: 0, R5: 0, R6: 0, D_ALU_BUD: 9, A_ALU_BUD: 10, MEM_BUS: 114, N: False, Z: False INDEX: 6, OPCODE: JMP, ARGS: [(absolute, 1, 0)]
  INFO    PC: 2, ADDR_BUS: 11, R1: 108, R2: 9, R3: 0, R4: 0, R5: 0, R6: 0, D_ALU_BUD: 9, A_ALU_BUD: 11, MEM_BUS: 108, N: False, Z: False INDEX: 1, OPCODE: LD, ARGS: [(register, r1, 0), (indirect, 2, r2)]
  INFO    PC: 3, ADDR_BUS: 11, R1: 108, R2: 9, R3: 0, R4: 0, R5: 0, R6: 0, D_ALU_BUD: 108, A_ALU_BUD: 11, MEM_BUS: 108, N: False, Z: False INDEX: 2, OPCODE: CMP, ARGS: [(register, r1, 0), (data, 0, 0)]
  INFO    PC: 4, ADDR_BUS: 11, R1: 108, R2: 9, R3: 0, R4: 0, R5: 0, R6: 0, D_ALU_BUD: 108, A_ALU_BUD: 11, MEM_BUS: 108, N: False, Z: False INDEX: 3, OPCODE: JZ, ARGS: [(absolute, 7, 0)]
  INFO    PC: 5, ADDR_BUS: 1, R1: 108, R2: 9, R3: 0, R4: 0, R5: 0, R6: 0, D_ALU_BUD: 108, A_ALU_BUD: 11, MEM_BUS: 108, N: False, Z: False INDEX: 4, OPCODE: ST, ARGS: [(absolute, 1, 0), (register, r1, 0)]
  INFO    PC: 6, ADDR_BUS: 1, R1: 108, R2: 10, R3: 0, R4: 0, R5: 0, R6: 0, D_ALU_BUD: 10, A_ALU_BUD: 11, MEM_BUS: 108, N: False, Z: False INDEX: 5, OPCODE: INC, ARGS: [(register, r2, 0)]
  INFO    PC: 1, ADDR_BUS: 1, R1: 108, R2: 10, R3: 0, R4: 0, R5: 0, R6: 0, D_ALU_BUD: 10, A_ALU_BUD: 11, MEM_BUS: 108, N: False, Z: False INDEX: 6, OPCODE: JMP, ARGS: [(absolute, 1, 0)]
  INFO    PC: 2, ADDR_BUS: 12, R1: 100, R2: 10, R3: 0, R4: 0, R5: 0, R6: 0, D_ALU_BUD: 10, A_ALU_BUD: 12, MEM_BUS: 100, N: False, Z: False INDEX: 1, OPCODE: LD, ARGS: [(register, r1, 0), (indirect, 2, r2)]
  INFO    PC: 3, ADDR_BUS: 12, R1: 100, R2: 10, R3: 0, R4: 0, R5: 0, R6: 0, D_ALU_BUD: 100, A_ALU_BUD: 12, MEM_BUS: 100, N: False, Z: False INDEX: 2, OPCODE: CMP, ARGS: [(register, r1, 0), (data, 0, 0)]
  INFO    PC: 4, ADDR_BUS: 12, R1: 100, R2: 10, R3: 0, R4: 0, R5: 0, R6: 0, D_ALU_BUD: 100, A_ALU_BUD: 12, MEM_BUS: 100, N: False, Z: False INDEX: 3, OPCODE: JZ, ARGS: [(absolute, 7, 0)]
  INFO    PC: 5, ADDR_BUS: 1, R1: 100, R2: 10, R3: 0, R4: 0, R5: 0, R6: 0, D_ALU_BUD: 100, A_ALU_BUD: 12, MEM_BUS: 100, N: False, Z: False INDEX: 4, OPCODE: ST, ARGS: [(absolute, 1, 0), (register, r1, 0)]
  INFO    PC: 6, ADDR_BUS: 1, R1: 100, R2: 11, R3: 0, R4: 0, R5: 0, R6: 0, D_ALU_BUD: 11, A_ALU_BUD: 12, MEM_BUS: 100, N: False, Z: False INDEX: 5, OPCODE: INC, ARGS: [(register, r2, 0)]
  INFO    PC: 1, ADDR_BUS: 1, R1: 100, R2: 11, R3: 0, R4: 0, R5: 0, R6: 0, D_ALU_BUD: 11, A_ALU_BUD: 12, MEM_BUS: 100, N: False, Z: False INDEX: 6, OPCODE: JMP, ARGS: [(absolute, 1, 0)]
  INFO    PC: 2, ADDR_BUS: 13, R1: 33, R2: 11, R3: 0, R4: 0, R5: 0, R6: 0, D_ALU_BUD: 11, A_ALU_BUD: 13, MEM_BUS: 33, N: False, Z: False INDEX: 1, OPCODE: LD, ARGS: [(register, r1, 0), (indirect, 2, r2)]
  INFO    PC: 3, ADDR_BUS: 13, R1: 33, R2: 11, R3: 0, R4: 0, R5: 0, R6: 0, D_ALU_BUD: 33, A_ALU_BUD: 13, MEM_BUS: 33, N: False, Z: False INDEX: 2, OPCODE: CMP, ARGS: [(register, r1, 0), (data, 0, 0)]
  INFO    PC: 4, ADDR_BUS: 13, R1: 33, R2: 11, R3: 0, R4: 0, R5: 0, R6: 0, D_ALU_BUD: 33, A_ALU_BUD: 13, MEM_BUS: 33, N: False, Z: False INDEX: 3, OPCODE: JZ, ARGS: [(absolute, 7, 0)]
  INFO    PC: 5, ADDR_BUS: 1, R1: 33, R2: 11, R3: 0, R4: 0, R5: 0, R6: 0, D_ALU_BUD: 33, A_ALU_BUD: 13, MEM_BUS: 33, N: False, Z: False INDEX: 4, OPCODE: ST, ARGS: [(absolute, 1, 0), (register, r1, 0)]
  INFO    PC: 6, ADDR_BUS: 1, R1: 33, R2: 12, R3: 0, R4: 0, R5: 0, R6: 0, D_ALU_BUD: 12, A_ALU_BUD: 13, MEM_BUS: 33, N: False, Z: False INDEX: 5, OPCODE: INC, ARGS: [(register, r2, 0)]
  INFO    PC: 1, ADDR_BUS: 1, R1: 33, R2: 12, R3: 0, R4: 0, R5: 0, R6: 0, D_ALU_BUD: 12, A_ALU_BUD: 13, MEM_BUS: 33, N: False, Z: False INDEX: 6, OPCODE: JMP, ARGS: [(absolute, 1, 0)]
  INFO    PC: 2, ADDR_BUS: 14, R1: 0, R2: 12, R3: 0, R4: 0, R5: 0, R6: 0, D_ALU_BUD: 12, A_ALU_BUD: 14, MEM_BUS: 0, N: False, Z: False INDEX: 1, OPCODE: LD, ARGS: [(register, r1, 0), (indirect, 2, r2)]
  INFO    PC: 3, ADDR_BUS: 14, R1: 0, R2: 12, R3: 0, R4: 0, R5: 0, R6: 0, D_ALU_BUD: 0, A_ALU_BUD: 14, MEM_BUS: 0, N: False, Z: True INDEX: 2, OPCODE: CMP, ARGS: [(register, r1, 0), (data, 0, 0)]
  INFO    PC: 7, ADDR_BUS: 14, R1: 0, R2: 12, R3: 0, R4: 0, R5: 0, R6: 0, D_ALU_BUD: 0, A_ALU_BUD: 14, MEM_BUS: 0, N: False, Z: True INDEX: 3, OPCODE: JZ, ARGS: [(absolute, 7, 0)]
